src/servers/src/http/event.rs

```diff
@@ -410,7 +408,7 @@ fn extract_pipeline_value_by_content_type(
 
 async fn ingest_logs_inner(
     state: LogHandlerRef,
-    pipeline_name: String,
+    pipeline_name: Option<String>,
```

Comment:
Using just `Option` to switch modes seems like a weak constraint, which can be unconsciously triggered by mistake. I was thinking of adding a bool flag to switch to 'auto-create' JSON mode by force, but that seems too heavy.
What do you think, @sunng87

Reply:
How about a enum:

```
PipelineChoice{
  UserDefined(String)
  Auto
```

Reply:
The question here is whether to add a parameter to specify the use of auto pipeline or to treat it as if it were used without the pipeline_name. or some other way to specify auto pipeline.

Reply:
On second thought, I think we could use `None` to indicate direct log write. This might usually happen in debugging or quick demo phases.

Do we need to validate the `Content-Type` to be JSON only in direct mode? If not, it would be the whole line as a column field and a `greptime_timestamp` as a timestamp index, which seems ok to me.

Reply:
I think we don't need to change the pipeline type to `Option`. The user must specify the pipeline name even when using the `identify` pipeline. We can ensure that the `identity` pipeline is the internal pipeline, and maybe we will add more internal pipelines in the future. Just checking if the pipeline name is reserved when the user creates new pipelines.

Reply:
I also favor the idea of defining some of the internal pipeline names. I've tentatively set all pipelines starting with `greptime_` to be for our internal use, and prohibited users from creating pipeline names starting with `greptime_`.